### ISSUE TITLE
Add: Support for Postgres

### ISSUE THREAD

**Issue by swhyte-rt** [moov-io/watchman#288]
Watchman Version: `current version and forward`

**What were you trying to do?**

Run watchman with Postgres as Database

I am just finishing off the implementation of this and can create a pull request.

**Comment by adamdecaf** [comment]
Yes please @swhyte-rt! Was it just adding the config options? We tried to keep the SQL compatible across the major databases.

**Comment by swhyte-rt** [comment]
It was adding config options, but also supporting differences in queries. For example Postgres did not like 

`insert into customer_status (customer_id, user_id, note, status, created_at) values (?, ?, ?, ?, ?);`

so I had to change to 

`insert into customer_status (customer_id, user_id, note, status, created_at) values ($1, $2, $3, $4, $5);` 

I will test this one more time, but it was throwing error when preparing the statement.

**Comment by swhyte-rt** [comment]
Here is the documentation explaining that most DB will support `?` syntax but Postgres supports `$1` and Oracle supports `:var1`

http://go-database-sql.org/prepared.html

**Comment by adamdecaf** [comment]
Ah that's too bad. Did you have to branch at each query off a config or flag?

**Comment by swhyte-rt** [comment]
I have repository interfaces, and have implemented a `generic<name of repo>Repository` (SQLite, MySQL) and a `postgres<name of repo>Repository`. It is in these implementation repos that is where syntax difference is.

**Comment by swhyte-rt** [comment]
@adamdecaf I can let you look at the fork of the work if you want to look at it before doing a pull request.

**Comment by adamdecaf** [comment]
Sure! I think it's this branch?

  https://github.com/moov-io/watchman/compare/master...swhyte-rt:master

**Comment by swhyte-rt** [comment]
Yep that is it

**Comment by swhyte-rt** [comment]
@adamdecaf I may have been over thinking this by breaking out separate repositories for generic db and Postgres. I was thinking if you were to ever support nosily databases. I can change this to a single repository with constants for the various queries and check the db type when initializing the query. That would be cleaner.

**Comment by adamdecaf** [comment]
We have no plans to support NoSQL for storage in Watchman. One repository for all three databases sounds fine so long as we find/replace and always prepare the query. 

I was checking with other projects and they've written an implementation for MySQL and Postgres.
- vault: [mysql](https://github.com/hashicorp/vault/blob/master/physical/mysql/mysql.go) and [postgres](https://github.com/hashicorp/vault/blob/master/physical/postgresql/postgresql.go)
- perkeep: [mysql](https://github.com/perkeep/perkeep/blob/master/pkg/sorted/mysql/mysqlkv.go) and [postgres](https://github.com/perkeep/perkeep/blob/master/pkg/sorted/postgres/postgreskv.go)

**Comment by swhyte-rt** [comment]
@adamdecaf I will look at those implementation and model after that.
